### PR TITLE
(FM-7790) Cleanup of Java_KS Test's

### PR DESCRIPTION
**Pull request by david22swan**
- Idempotent Manifest Apply Implemented
- ServerSpec Test Facts Implemented
- Duplicate Code Removed and Test's Folded Together

### CODE REVIEW

File: spec/acceptance/keystore_spec.rb [hunk 1]
```diff
@@ -57,42 +52,65 @@
...
         MANIFEST
 
-        apply_manifest(pp, catch_failures: true)
-        apply_manifest(pp, catch_changes: true)
+        idempotent_apply(default, pp)
       end
 
       expectations = [
-        %r{Your keystore contains 2 entries},
+        %r{Your keystore contains 1 entry},
```
Comment (tphoney) [inline spec/acceptance/keystore_spec.rb]: how come there is only a single entry now ?

File: spec/acceptance/chain_key_spec.rb [hunk 2]
```diff
@@ -182,30 +147,4 @@
       it { is_expected.not_to be_file }
     end
   end
-
-  describe 'managing existing java chain keys in noop', unless: UNSUPPORTED_PLATFORMS.include?(fact('operatingsystem')) do
```
Comment (tphoney) [inline spec/acceptance/chain_key_spec.rb]: have these tests moved up or are not needed ?

**Comment by tphoney** [comment]
looks really good. the changes in  spec/acceptance/chain_key_spec.rb are hard to follow. how come some of the tests have been removed. have they moved / not needed. what is the reasoning

File: spec/acceptance/keystore_spec.rb (hunk 1 above)
Comment (david22swan) [reply spec/acceptance/keystore_spec.rb]: The password fails test wipes out the second keystore puppetca2 which was previously left behind. I double checked the stdout and it look's correct.

File: spec/acceptance/chain_key_spec.rb (hunk 2 above)
Comment (david22swan) [reply spec/acceptance/chain_key_spec.rb]: I folded them up into the first one in order to cut down on the number of manifest applies

Comment (david22swan) [reply spec/acceptance/chain_key_spec.rb]: Or at least the first one, did not move this one. No idea why it is showing up red, It's still in place in my local copy and when viewing the file it has not disappeared.

Comment (tphoney) [reply spec/acceptance/chain_key_spec.rb]: so the number of manifests run, has dropped from 6 to 3. is that because you have grouped 2 tests in one ?

Comment (david22swan) [reply spec/acceptance/chain_key_spec.rb]: From 5 to 3, but yeh. Should update the test description though. Look's like it doesn't really match. A failure just appeared on Ubuntu 18.04 on adhoc so will do it a the same time as I fix it.

**Comment by david22swan** [comment]
Screen Shot 2019-03-14 at 11 27 40 AM
https://jenkins-master-prod-1.delivery.puppetlabs.net/view/modules/view/cross-platform/view/_adhoc/view/java_ks/job/forge-module_puppetlabs-java_ks_init-manual-parameters_adhoc/24/
Adhoc Passed
